pkg/xds/bootstrap/generator_test.go

```diff
@@ -218,4 +218,121 @@ var _ = Describe("bootstrapGenerator", func() {
 		// expect
 		Expect(actual).To(MatchYAML(expected))
 	})
+
+	It("should fail bootstrap configuration due to conflicting port in inbound", func() {
+		// setup
+		dataplane := mesh.DataplaneResource{
+			Spec: mesh_proto.Dataplane{
+				Networking: &mesh_proto.Dataplane_Networking{
+					Address: "8.8.8.8",
+					Inbound: []*mesh_proto.Dataplane_Networking_Inbound{
+						{
+							Address:     "127.0.0.111",
+							Port:        9901,
+							ServicePort: 8443,
+							Tags: map[string]string{
+								"service": "backend",
+							},
+						},
+					},
+					Outbound: []*mesh_proto.Dataplane_Networking_Outbound{
+						{
+							Address: "1.1.1.1",
+							Port:    9000,
+							Service: "redis",
+						},
+					},
+				},
+			},
+		}
+		// when
+		err := resManager.Create(context.Background(), &dataplane, store.CreateByKey("name-1.namespace", "mesh"))
+		// then
+		Expect(err).ToNot(HaveOccurred())
+
+		// when
+		dataplane.Spec.Networking.Address = "127.0.1.123"
+		dataplane.Spec.Networking.Inbound[0].Address = "2.2.2.2"
+		err = resManager.Create(context.Background(), &dataplane, store.CreateByKey("name-2.namespace", "mesh"))
+		// then
+		Expect(err).ToNot(HaveOccurred())
+
+		// given
+		params := bootstrap_config.DefaultBootstrapParamsConfig()
+		params.XdsHost = "127.0.0.1"
+		params.XdsPort = 5678
+
+		generator := NewDefaultBootstrapGenerator(resManager, params, "")
+		request := types.BootstrapRequest{
+			Mesh:      "mesh",
+			Name:      "name-1.namespace",
+			AdminPort: 9901,
+		}
+
+		// when
+		_, err = generator.Generate(context.Background(), request)
+		// then
+		Expect(err).To(HaveOccurred())
+
+		request = types.BootstrapRequest{
+			Mesh:      "mesh",
+			Name:      "name-2.namespace",
+			AdminPort: 9901,
+		}
+
+		// when
+		_, err = generator.Generate(context.Background(), request)
+		// then
+		Expect(err).To(HaveOccurred())
+
+	})
+
+	It("should fail bootstrap configuration due to conflicting port in outbound", func() {
+		// setup
+		dataplane := mesh.DataplaneResource{
+			Spec: mesh_proto.Dataplane{
+				Networking: &mesh_proto.Dataplane_Networking{
+					Address: "8.8.8.8",
+					Inbound: []*mesh_proto.Dataplane_Networking_Inbound{
+						{
+							Port:        443,
+							ServicePort: 8443,
+							Tags: map[string]string{
+								"service": "backend",
+							},
+						},
+					},
+					Outbound: []*mesh_proto.Dataplane_Networking_Outbound{
+						{
+							Address: "127.0.1.1",
```

Comment:
I think I have some blinds spots about networking. Why `127.0.1.1:9901` and `127.0.0.1:9901` is a conflict?

Reply:
Right, I was reading  RFC 5735, Section 3.
```
   127.0.0.0/8 - This block is assigned for use as the Internet host
   loopback address.  A datagram sent by a higher-level protocol to an
   address anywhere within this block loops back inside the host.
```
And kind of thought that in the end there will be a conflict. However, at least the Linux TCP/IP stack distinguishes between he IP addresses and allows for binding to different IP's in that range using the same port. Therefore - there is no conflict in this situation. Let me fix that.